deepchecks/checks/methodology/model_inference_time.py

```diff
@@ -0,0 +1,94 @@
+"""The model inference time check module."""
+import typing as t
+import timeit
+
+import pandas as pd
+
+from deepchecks import SingleDatasetBaseCheck, CheckResult, Dataset, ConditionResult
+from deepchecks.utils import model_type_validation
+from deepchecks.string_utils import format_number
+
+
+__all__ = ['ModelInferenceTimeCheck']
+
+
+MI = t.TypeVar('MI', bound='ModelInferenceTimeCheck')
+
+
+class ModelInferenceTimeCheck(SingleDatasetBaseCheck):
+    """Measure model average inference time (in seconds) per sample.
+
+    Args:
+        number_of_samples (int): number of samples to use for inference
+    """
+
+    def __init__(self, number_of_samples: int = 1000):
+        self.number_of_samples = number_of_samples
+        super().__init__()
+
+    def run(self, dataset: Dataset, model: object) -> CheckResult:
+        """Run check.
+
+        Args:
+            dataset (Dataset): samples that will be used to measure inference time
+            model (BaseEstimator): a model to measure inference time
+
+        Returns:
+            CheckResult:
+                value is of the type 'float'
+
+        Raises:
+            DeepchecksValueError: If the 'test_dataset' is not a 'Dataset' instance with a label or
+                if 'model' is not a scikit-learn-compatible fitted estimator instance
+        """
+        return self._model_inference_time_check(dataset, model)
+
+    def _model_inference_time_check(
+        self,
+        dataset: Dataset,
+        model: object # TODO: find more precise type for model
+    ) -> CheckResult:
+        check_name = type(self).__name__
+        Dataset.validate_dataset(dataset, check_name)
+        Dataset.validate_model(dataset, model)
+        Dataset.validate_features(dataset, check_name)
+        model_type_validation(model)
+
+        prediction_method = model.predict # type: ignore
+        df = t.cast(pd.DataFrame, dataset.features_columns())
+
+        number_of_samples = len(df) if len(df) < self.number_of_samples else self.number_of_samples
+        df = df[:number_of_samples]
```

Comment:
use dataframe sample method to take random samples (with random_state as seed)

Reply:
updated